examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,650 @@
+        return pw::OkStatus();
+    }
+
+    void StartScan(ServerContext &, const chip_rpc_ScanConfig & request, ServerWriter<chip_rpc_ScanResults> & writer)
+    {
+        // Unregister any previous listeners so that no false
+        // WIFI_EVENT_SCAN_DONE event is triggered.
+        esp_err_t err = esp_event_handler_unregister(WIFI_EVENT, WIFI_EVENT_SCAN_DONE, &scan_event_handler);
+        if (ESP_OK != err)
+        {
+            ESP_LOGE(TAG, "Cannot unregister previous scan event handler, error: %d", err);
+            return;
+        }
+        scanresult_writer = std::move(writer);
+        wifi_scan_config_t scan_config{ 0 };
+        if (request.ssid_count != 0)
+        {
+            scan_config.ssid = const_cast<uint8_t *>(reinterpret_cast<const uint8_t *>(request.ssid[0].bytes));
+        }
+        if (request.bssid_count != 0)
+        {
+            scan_config.bssid = const_cast<uint8_t *>(reinterpret_cast<const uint8_t *>(request.bssid[0].bytes));
+        }
+        scan_config.channel     = request.channel;
+        scan_config.show_hidden = request.show_hidden;
+        scan_config.scan_type   = static_cast<wifi_scan_type_t>(request.active_scan);
+        if (request.active_scan)
+        {
+            scan_config.scan_time.active.min = request.scan_time_min_ms;
+            scan_config.scan_time.active.max = request.scan_time_max_ms;
+        }
+        else
+        {
+            scan_config.scan_time.passive = request.scan_time_min_ms;
+        }
+        esp_wifi_scan_start(&scan_config, false /* block */);
+        // Scan results will be sent to scanresult_writer once
+        // WIFI_EVENT_SCAN_DONE event is received
+        err = esp_event_handler_register(WIFI_EVENT, WIFI_EVENT_SCAN_DONE, &scan_event_handler, NULL);
+    }
+
+    pw::Status StopScan(ServerContext &, const chip_rpc_Empty & request, chip_rpc_Empty & response)
+    {
+        esp_wifi_scan_stop();
+        return pw::OkStatus();
+    }
+
+    pw::Status Connect(ServerContext &, const chip_rpc_ConnectionData & request, chip_rpc_ConnectionResult & response)
+    {
+        wifi_config_t wifi_config {
+        .sta = {
+            /* Setting a password implies station will connect to all security modes including WEP/WPA.
+             * However these modes are deprecated and not advisable to be used. Incase your Access point
+             * doesn't support WPA2, these mode can be enabled by commenting below line */
+	        .threshold = {
+                .authmode = static_cast<wifi_auth_mode_t>(request.security_type),
+            },
+
+            .pmf_cfg = {
+                .capable = true,
+                .required = false
+            },
+        },
+    };
+        memcpy(wifi_config.sta.ssid, request.ssid.bytes,
+               std::min(sizeof(wifi_config.sta.ssid), static_cast<size_t>(request.ssid.size)));
+        memcpy(wifi_config.sta.password, request.secret, std::min(sizeof(wifi_config.sta.password), strlen(request.secret)));
```

Comment:
cannot use strlen, since Wifi password test cases may contain 0 bytes within the passphrase portion (since passphrases are octet strings). Also, does that assume there is null-termination in the protobuf wire strings?

Reply:
I changed the proto to use a byte array instead of a string.